File: NC/src/ForwardDifference.cpp
#include "../include/ForwardDifference.hpp"

ForwardDifference::ForwardDifference(function<double(double)> f, double h)
    : Differentiation(f, h) {}

double ForwardDifference::derivative(double x) const {
    return (func(x + h) - func(x)) / h;
}

File: Numerical_Computing_2/src/differentiation/forward_difference.py
from .numerical_differentiation import NumericalDifferentiation


class ForwardDifference(NumericalDifferentiation):
    def derivative(self, x: float) -> float:
        return (self.func(x + self.h) - self.func(x)) / self.h

    @property
    def method_name(self) -> str:
        return "Forward Difference"
